lib/component.js

```diff
@@ -362,8 +362,20 @@ class Component extends WebComponent {
     }
   }
 
-  logError() {
-    console.error(...arguments);
+  logRenderError(message, error) {
+    console.error(message, this, `\n`, error);
+  }
+
+  emitRenderError(message, error) {
+    this.el.dispatchEvent(new CustomEvent(`rendererror`, {
+      detail: {message, error},
+      bubbles: true, composed: true,
+    }));
+  }
+
+  onRenderError(message, error) {
+    this.logRenderError(message, error);
+    this.emitRenderError(message, error);
```

Comment:
This division into 3 single-use functions seems a bit needlessly fussy. Also if it's going to be specifically about errors during rendering then the `message` param is superfluous. How about
```js
onRenderError(error) {
  const message = `Error while rendering ${this.toString()}`;
  console.error(message, this, `\n`, error);
  this.el.dispatchEvent(new CustomEvent(`rendererror`, {
    detail: {message, error},
    bubbles: true, composed: true,
  }));
}
```

Reply:
the only reason I split it is because the test framework/headless browser fails the test on `console.error`. the test spy was also a stub function before and prevented the call